packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte

```diff
@@ -117,15 +145,248 @@
       }
     }
   }
-  const onChange = Utils.sequential(async (e, key) => {
+
+  $: customStepLayouts($memoBlock, schemaProperties)
+
+  const customStepLayouts = block => {
+    if (
+      rowSteps.includes(block.stepId) ||
+      (rowTriggers.includes(block.stepId) && isTestModal)
+    ) {
+      const schema = schemaProperties.reduce((acc, entry) => {
+        const [key, val] = entry
+        acc[key] = val
+        return acc
+      }, {})
+
+      // Optionally build the rev field config when its needed.
+      const getRevConfig = () => {
+        const rowRevEntry = schema["revision"]
+        if (!rowRevEntry) {
+          return []
+        }
+        const rowRevlabel = getFieldLabel("revision", rowRevEntry)
+
+        return isTestModal
+          ? [
+              {
+                type: DrawerBindableInput,
+                title: rowRevlabel,
+                props: {
+                  panel: AutomationBindingPanel,
+                  value: inputData["revision"],
+                  onChange: e => {
+                    onChange({ ["revision"]: e.detail })
+                  },
+                  bindings,
+                  updateOnChange: false,
+                  forceModal: true,
+                },
+              },
+            ]
+          : []
+      }
+
+      const getIdConfig = () => {
+        const rowIdentifier = isTestModal ? "id" : "rowId"
+
+        const rowIdEntry = schema[rowIdentifier]
+        if (!rowIdEntry) {
+          return []
+        }
+
+        const rowIdlabel = getFieldLabel(rowIdentifier, rowIdEntry)
+
+        return [
+          {
+            type: DrawerBindableInput,
+            title: rowIdlabel,
+            props: {
+              panel: AutomationBindingPanel,
+              value: inputData[rowIdentifier],
+              onChange: e => {
+                onChange({ [rowIdentifier]: e.detail })
+              },
+              bindings,
+              updateOnChange: false,
+              forceModal: true,
+            },
+          },
+        ]
+      }
+
+      stepLayouts[block.stepId] = {
```

Comment:
@deanhannigan Hi Dean, see for this, there was some work done recently to expose the `oldRow` within the frontend. This is an input for the Update Row trigger as well so the user needs to be able to add it within their Test Data. I think it was just generated out of the trigger schema before but doesn't seem to be there now, what do you think the best way to handle this is?

Reply:
Hey Peter,  good spot! The layout for the screen has its own configuration and its the only step type that does this at the minute. The UX just needs to be configured and mapped to the correct location in the testData. I'll have a look at this now 👍